packages/debugger/app/hooks/use-xmtp-identity.tsx

```diff
@@ -0,0 +1,174 @@
+            },
+            signMessage(message) {
+              return signMessage(config, {
+                message: {
+                  raw:
+                    typeof message === "string"
+                      ? Buffer.from(message)
+                      : Buffer.from(message as Uint8Array),
+                },
+              });
+            },
+          }
+        : null,
+    [address, config]
+  );
+
+  function getSignerFromLocalStorage() {
+    if (typeof window !== "undefined") {
+      const storedData = localStorage.getItem(LOCAL_STORAGE_KEYS.XMTP_SIGNER);
+      if (storedData) {
+        const signerRaw = JSON.parse(storedData);
+        const signer: XmtpSigner = {
+          walletAddress: signerRaw.walletAddress,
+          keys: Buffer.from(signerRaw.keys, "hex"),
+        };
+        return signer;
+      }
+      return null;
+    }
+
+    return null;
+  }
+
+  useEffect(() => {
+    const signer = getSignerFromLocalStorage();
+    if (signer) setXmtpSigner(signer);
+  }, []);
+
+  useEffect(() => {
+    if (xmtpSigner) {
+      Client.create(null, {
+        privateKeyOverride: xmtpSigner.keys,
+      }).then((client) => setXmtpClient(client));
+    }
+  }, [xmtpSigner]);
+
+  function logout() {
+    localStorage.removeItem(LOCAL_STORAGE_KEYS.XMTP_SIGNER);
+    setXmtpSigner(null);
+  }
+
+  async function onSignerlessFramePress() {
+    setLoading(true);
+    await createAndStoreSigner();
+    setLoading(false);
+  }
+
+  async function createAndStoreSigner() {
+    try {
+      if (!xmtpSigner) {
+        if (!walletSigner) {
+          connect.openConnectModal?.();
+          return;
+        }
+        Client.getKeys(walletSigner, {
+          env: "dev",
+          skipContactPublishing: true,
+          persistConversations: false,
+        }).then((keys) => {
+          const walletAddress = getAccount(config).address || zeroAddress;
+          localStorage.setItem(
+            LOCAL_STORAGE_KEYS.XMTP_SIGNER,
+            JSON.stringify({
+              walletAddress,
+              keys: Buffer.from(keys).toString("hex"),
+            })
+          );
+          setXmtpSigner({
+            keys,
+            walletAddress,
+          });
+        });
+      }
+    } catch (error) {
+      console.error("frames.js: API Call failed", error);
+    }
+  }
+
+  return {
+    signer: xmtpSigner,
+    hasSigner: !!xmtpSigner?.keys,
+    async signFrameAction(actionContext) {
+      if (!xmtpClient) {
+        throw new Error("No xmtp client");
+      }
+
+      if (!address) {
+        throw new Error("No address");
+      }
+
+      const framesClient = new FramesClient(xmtpClient);
+      const payload = await framesClient.signFrameAction({
+        frameUrl: actionContext.url,
+        inputText: actionContext.inputText,
+        state: actionContext.state,
+        buttonIndex: actionContext.buttonIndex,
+        conversationTopic:
+          actionContext.frameContext.conversationTopic || "/xmtp/0/123",
```

Comment:
what does `/xmtp/0/123` mean? Shouldn't we require conversation topic? Asking just to understand the logic.